lwm2m-client-flow-access-object.c

```diff
@@ -56,281 +57,229 @@
+		return 1;
+	}
 
-		case FLOWM2M_FLOW_ACCESS_OBJECT_REMEMBERMETOKEN:
-			if (flowAccessObject.RememberMeToken)
-				free(flowAccessObject.RememberMeToken);
-			flowAccessObject.RememberMeToken = (char *)malloc(srcBufferLen + 1);
-			memset(flowAccessObject.RememberMeToken, 0, srcBufferLen + 1);
-			memcpy(flowAccessObject.RememberMeToken, srcBuffer, srcBufferLen);
-			result = srcBufferLen;
-			break;
+	error = AwaStaticClient_DefineResourceWithHandler(awaClient, "CustomerSecret", 20001, 2, AwaResourceType_String, 0, 1, AwaResourceOperations_ReadWrite,
+		accessHandler);
+	if (error != AwaError_Success)
+	{
+		printf("Failed to define CustomerSecret resource\n");
+		return 1;
+	}
 
-		case FLOWM2M_FLOW_ACCESS_OBJECT_REMEMBERMETOKENEXPIRY:
-			memcpy(&flowAccessObject.RememberMeTokenExpiry, srcBuffer, srcBufferLen);
-			result = srcBufferLen;
-			break;
+	error = AwaStaticClient_DefineResourceWithHandler(awaClient, "RememberMeToken", 20001, 3, AwaResourceType_String, 0, 1, AwaResourceOperations_ReadWrite,
+		accessHandler);
+	if (error != AwaError_Success)
+	{
+		printf("Failed to define RememberMeToken resource\n");
+		return 1;
+	}
 
-		default:
-			result = -1;
-			break;
+	error = AwaStaticClient_DefineResourceWithHandler(awaClient, "RememberMeTokenExpiry", 20001, 4, AwaResourceType_Opaque, 0, 1, AwaResourceOperations_ReadWrite,
+		accessHandler);
+	if (error != AwaError_Success)
+	{
+		printf("Failed to define RememberMeTokenExpiry resource\n");
+		return 1;
 	}
 
-	return result;
+	return 0;
 }
 
-/***************************************************************************************************
- * Implementation - Public
- **************************************************************************************************/
-
-int Lwm2m_RegisterFlowAccessObject(Lwm2mContextType * context)
+int CreateFlowAccessObject(AwaStaticClient *awaClient)
 {
-	Lwm2mCore_RegisterObjectType(context, "FlowAccess" , FLOWM2M_FLOW_ACCESS_OBJECT,
-		MultipleInstancesEnum_Single, MandatoryEnum_Optional, &flowAccessObjectOperationHandlers);
-
-	REGISTER_FLOW_ACCESS_OBJECT_RESOURCE(context, "URL", FLOWM2M_FLOW_ACCESS_OBJECT_URL, \
-		ResourceTypeEnum_TypeString);
-	REGISTER_FLOW_ACCESS_OBJECT_RESOURCE(context, "CustomerKey", \
-		FLOWM2M_FLOW_ACCESS_OBJECT_CUSTOMERKEY, ResourceTypeEnum_TypeString);
-	REGISTER_FLOW_ACCESS_OBJECT_RESOURCE(context, "CustomerSecret", \
-		FLOWM2M_FLOW_ACCESS_OBJECT_CUSTOMERSECRET, ResourceTypeEnum_TypeString);
-	REGISTER_FLOW_ACCESS_OBJECT_RESOURCE(context, "RememberMeToken", \
-		FLOWM2M_FLOW_ACCESS_OBJECT_REMEMBERMETOKEN, ResourceTypeEnum_TypeString);
-	REGISTER_FLOW_ACCESS_OBJECT_RESOURCE(context, "RememberMeTokenExpiry", \
-		FLOWM2M_FLOW_ACCESS_OBJECT_REMEMBERMETOKENEXPIRY, ResourceTypeEnum_TypeOpaque);
+	AwaError error;
 
+	error = AwaStaticClient_CreateObjectInstance(awaClient, FLOWM2M_FLOW_ACCESS_OBJECT, 0);
+	if (error != AwaError_Success)
+	{
+		printf("failed to create flow access object instance\n");
```

Comment:
Make first letter of any debug print Capital to maintain the consistency.
failed -> Failed